packages/dev/src/commands/watch.js

```diff
@@ -0,0 +1,77 @@
+import { spawn } from 'node:child_process';
+import { join } from 'node:path';
+
+import { getClerkPackages } from '../utils/getClerkPackages.js';
+import { getDependencies } from '../utils/getDependencies.js';
+import { getMonorepoRoot } from '../utils/getMonorepoRoot.js';
+
+/**
+ * Starts long-running watchers for Clerk dependencies.
+ * @param {object} args
+ * @param {boolean | undefined} args.js If `true`, only spawn the builder for `@clerk/clerk-js`.
+ * @returns {Promise<void>}
+ */
+export async function watch({ js }) {
+  const { dependencies, devDependencies } = await getDependencies(join(process.cwd(), 'package.json'));
+  const clerkPackages = Object.keys(await getClerkPackages());
+
+  const packagesInPackageJSON = [...Object.keys(dependencies ?? {}), ...Object.keys(devDependencies ?? {})];
+  const clerkPackagesInPackageJSON = packagesInPackageJSON.filter(p => clerkPackages.includes(p));
+
+  const filterArgs = clerkPackagesInPackageJSON.map(p => `--filter=${p}`);
+
+  const args = ['watch', 'build', ...filterArgs];
```

Comment:
According to https://turbo.build/repo/docs/reference/watch#using-turbo-watch-with-persistent-tasks

> When your script has a built-in watcher, you likely don't need to use turbo watch. Instead, use your script's built-in watcher and mark the task as long-running using ["persistent": true](https://turbo.build/repo/docs/reference/configuration#persistent).

Can't we use the `dev` script in each package instead? And adjust the `dev` script of `clerk-js` to set the `devOrigin`?

Then the need for a separate window to run `turbo run dev --filter=@clerk/clerk-js -- --env devOrigin=http://localhost:4000"` would be removed

Reply:
Using `dev` was my initial approach, but I ran into a lot of issues where the `dev` commands ran in parallel even for dependent packages. To alleviate this, we use `turbo watch`, which understands the order that the `build` operations need to execute in for the packages to build correctly.